pkg/plugins/runtime/k8s/probes/probe.go

```diff
@@ -0,0 +1,53 @@
+package probes
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	kube_core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type KumaProbe kube_core.Probe
+
+func (p KumaProbe) ToInbound(port uint32) (KumaProbe, bool) {
```

Comment:
It is kind of hard for me to figure out what this method does.

How about `ToReal` and explain in the comment on `KumaProbe` what is the real probe and what is a virtual probe and why do we even have this.

on `ToReal` comment that when executed on virtual probe it converts back to real probe. And when executed on real probe it will return false (?). Shouldn't this bool we an error?